### ISSUE TITLE
Update Consul Core goiang k8s submodules to match Consul K8s

### ISSUE THREAD

**Issue by david-yu** [hashicorp/consul#10458]
#### Feature Description

As of Consul 1.10 we have the following module versions in [consul-k8s](https://github.com/hashicorp/consul-k8s/blob/7548958a26873ef45e1a21e997ce03282ab1236e/go.mod#L36)
```
	k8s.io/api v0.21.1
	k8s.io/apimachinery v0.21.1
	k8s.io/client-go v0.21.1
```
Current go mods as of Consul 1.10 are as follows:
```
	k8s.io/api v0.16.9
	k8s.io/apimachinery v0.16.9
	k8s.io/client-go v0.16.9
```
#### Use Case(s)

Any relevant use-cases that you see.

**Comment by dnephin** [comment]
Unfortunately this update brings in a newer version of protobuf which is likely not an easy upgrade ([1](https://github.com/hashicorp/consul/pull/10466#pullrequestreview-690831756), [2](https://github.com/hashicorp/consul/pull/9030)).

Does this version mismatch cause any problems?

**Comment by david-yu** [comment]
It does not cause any problems right now in Consul K8s, it does come up as a warning in a security scan but this is not a high priority issue right now.

**Comment by rboyer** [comment]
Blocked on https://github.com/hashicorp/consul/issues/10471

**Comment by mikemorris** [comment]
We've gotten a bit closer here after merging https://github.com/hashicorp/consul/pull/10898, which bumps us to k8s.io/api 0.18.2

**Comment by david-yu** [comment]
@rboyer Out of curiosity, how hard is it to bump to the latest versions of the k8s modules that we have in `consul-k8s` now the gogo/protobuf dragons have been defeated? I believe this is last of our medium severity cve scans, although this is not invoked directly. Current versions we use are ([link](https://github.com/hashicorp/consul-k8s/blob/5e4cabf0f36c0a5f764340147c19c0b9ae6804c8/control-plane/go.mod#L23))

```
	k8s.io/api v0.22.2
	k8s.io/apimachinery v0.22.2
	k8s.io/client-go v0.22.2
```

**Comment by david-yu** [comment]
Oh ok got it. Yea I believe as long as we get to 0.20.0 we are in the clear for these dependencies. Mainly trying to address optics around security scans as it may require security exceptions in some very security conscious environments. 

```
	k8s.io/api v0.22.2
	k8s.io/apimachinery v0.22.2
	k8s.io/client-go v0.22.2
```

**Comment by david-yu** [comment]
If not for 1.12.x would it be possible to consider just for 1.13.x?

**Comment by david-yu** [comment]
Yes that would be much appreciated, 0.22.2 is nice to have but at least those versions are required to get us past the CVE.
